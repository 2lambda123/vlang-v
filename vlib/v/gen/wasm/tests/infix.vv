fn mul(a i64, unsigned u32) f64 {
	mut one := a

	one *= 2 / unsigned

	return one
}

fn typ(a int) i64 {
	mut one := a
	sec := one + i64(a)

	return sec
}
