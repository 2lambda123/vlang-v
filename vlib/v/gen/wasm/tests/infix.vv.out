(module
 (type $i64_i32_=>_f64 (func (param i64 i32) (result f64)))
 (type $i32_=>_i64 (func (param i32) (result i64)))
 (type $none_=>_none (func))
 (func $main.mul (param $0 i64) (param $1 i32) (result f64)
  (local $2 i64)
  local.get $0
  local.set $2
  local.get $2
  i32.const 2
  local.get $1
  i32.div_u
  i64.extend_i32_u
  i64.mul
  local.set $2
  local.get $2
  f64.convert_i64_s
  return
 )
 (func $main.typ (param $0 i32) (result i64)
  (local $1 i32)
  (local $2 i64)
  local.get $0
  local.set $1
  local.get $1
  local.get $0
  i64.extend_i32_s
  i32.wrap_i64
  i32.add
  i64.extend_i32_s
  local.set $2
  local.get $2
  return
 )
 (func $main.main
  nop
 )
)